cfgmgr/intfmgr.cpp

```diff
@@ -71,51 +98,139 @@ bool IntfMgr::isIntfStateOk(const string &alias)
 
     return false;
 }
-void IntfMgr::doTask(Consumer &consumer)
+
+bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
+        const vector<FieldValueTuple>& data,
+        const string& op)
 {
     SWSS_LOG_ENTER();
 
-    auto it = consumer.m_toSync.begin();
-    while (it != consumer.m_toSync.end())
+    string alias(keys[0]);
+    string vrf_name = "";
+
+    for (auto idx : data)
     {
-        KeyOpFieldsValuesTuple t = it->second;
+        const auto &field = fvField(idx);
+        const auto &value = fvValue(idx);
+        if (field == "vnet_name" || field == "vrf_name")
+        {
+            vrf_name = value;
+        }
+    }
 
-        vector<string> keys = tokenize(kfvKey(t), config_db_key_delimiter);
+    if (op == SET_COMMAND)
+    {
+        /*
+         * Don't proceed if port/LAG/VLAN is not ready yet.
+         * The pending task will be checked periodically and retried.
+         * TODO: Subscribe to stateDB for port/lag/VLAN state and retry
+         * pending tasks immediately upon state change.
+         */
+        if (!isIntfStateOk(alias))
+        {
+            SWSS_LOG_DEBUG("Interface is not ready, skipping %s", alias.c_str());
+            return false;
+        }
 
-        if (keys.size() != 2)
+        if (!isIntfStateOk(vrf_name))
         {
-            SWSS_LOG_ERROR("Invalid key %s", kfvKey(t).c_str());
-            it = consumer.m_toSync.erase(it);
-            continue;
+            SWSS_LOG_DEBUG("VRF is not ready, skipping %s", vrf_name.c_str());
+            return false;
+        }
+
+        setIntfVrf(alias, vrf_name);
+        m_appIntfTableProducer.set(alias, data);
+    }
+    else if (op == DEL_COMMAND)
+    {
+        setIntfVrf(alias, "");
+        m_appIntfTableProducer.del(alias);
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Unknown operation: %s", op.c_str());
+    }
+
+    return true;
+}
+
+bool IntfMgr::doIntfAddrTask(const vector<string>& keys,
+        const vector<FieldValueTuple>& data,
+        const string& op)
+{
+    SWSS_LOG_ENTER();
+
+    string alias(keys[0]);
+    IpPrefix ip_prefix(keys[1]);
+    string appKey = keys[0] + ":" + keys[1];
+
+    if (op == SET_COMMAND)
+    {
+        /*
+         * Don't proceed if port/LAG/VLAN is not ready yet.
+         * The pending task will be checked periodically and retried.
+         * TODO: Subscribe to stateDB for port/lag/VLAN state and retry
+         * pending tasks immediately upon state change.
+         */
+        if (!isIntfStateOk(alias))
+        {
+            SWSS_LOG_DEBUG("Interface is not ready, skipping %s", alias.c_str());
+            return false;
         }
 
-        string alias(keys[0]);
-        IpPrefix ip_prefix(keys[1]);
+        setIntfIp(alias, "add", ip_prefix.to_string(), ip_prefix.isV4());
```

Comment:
We have to check for return value here. Previously it was via Netlink and hence ensured it was configured in kernel. Now, we have to explicitly check if the address is successfully configured in kernel

Reply:
Done

Reply:
Using EXEC_WITH_ERROR_THROW